src/dune_rules/preprocessing.ml

```diff
@@ -710,6 +710,10 @@ let pp_one_module sctx ~lib_name ~scope ~preprocessor_deps
                           ; Path (Path.build dst)
                           ; Command.Ml_kind.ppx_driver_flag ml_kind
                           ; Dep (Path.build src)
+                          ; Hidden_deps
```

Comment:
I don't understand why this is needed. 

IIUC, the pipeline looks something like:

```
.re -> (dialect) .re.ml -> (pps) .pp.ml
```

So isn't the original source file already a (transitive) dependency of the pps action?

```

Reply:
But sandboxing does not take the transitive closure, onoy the specified deoendencies. Thats why its important to always fully specify them.